pkg/scheduler/plugins/predicates/predicates.go

```diff
@@ -525,15 +552,29 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 			proportionalStatus, _ := checkNodeResourceIsProportional(task, node, predicate.proportional)
 			if proportionalStatus.Code != api.Success {
 				predicateStatus = append(predicateStatus, proportionalStatus)
-				return api.NewFitErrWithStatus(task, node, predicateStatus...)
+				if shouldFailPredication(proportionalStatus) {
+					return api.NewFitErrWithStatus(task, node, predicateStatus...)
+				}
 			}
 			klog.V(4).Infof("checkNodeResourceIsProportional predicates Task <%s/%s> on Node <%s>: fit %v",
 				task.Namespace, task.Name, node.Name, fit)
 		}
+
+		if len(predicateStatus) > 0 {
+			return api.NewFitErrWithStatus(task, node, predicateStatus...)
+		}
+
 		return nil
 	})
 }
 
+func shouldFailPredication(status *api.Status) bool {
```

Comment:
A redundant logic with:
https://github.com/volcano-sh/volcano/blob/7170cca5a0af67938a05f46f3bf5095a127b13b9/pkg/scheduler/framework/session.go#L426
And can we just check there?

Reply:
The early return added to the predicate function is to prevent executing unnecessary operations further down the logic.

Additionally, the logic in the diagram evaluates stateSets, and I believe shouldFailPredication is an abstraction tailored to the needs of the predicate plugin itself. Its logic may change in the future, so wrapping it within the predicate function is a better choice.